kotlin-inject-compiler/core/src/main/kotlin/me/tatarka/inject/compiler/KmpComponentCreateGenerator.kt

```diff
@@ -1,20 +1,39 @@
 package me.tatarka.inject.compiler
 
+import com.squareup.kotlinpoet.AnnotationSpec
+import com.squareup.kotlinpoet.ClassName
 import com.squareup.kotlinpoet.CodeBlock
 import com.squareup.kotlinpoet.FileSpec
 import com.squareup.kotlinpoet.FunSpec
 import com.squareup.kotlinpoet.KModifier
-import me.tatarka.kotlin.ast.AstClass
-import me.tatarka.kotlin.ast.AstFunction
+import com.squareup.kotlinpoet.TypeName
 import me.tatarka.kotlin.ast.AstProvider
 
+data class ComponentClassInfo(
+    val packageName: String,
+    val name: String,
+    val companionClassName: ClassName?,
+    val typeName: TypeName,
+    val className: ClassName,
+)
+
+data class KmpComponentCreateFunctionInfo(
+    val name: String,
+    val annotations: List<AnnotationSpec>,
+    val visibility: KModifier,
+    val receiverParameterType: TypeName?,
+    val parameters: List<Pair<String, TypeName>>,
+    val parametersTemplate: String,
+    val parameterNames: List<String>,
+)
```

Comment:
I agree. The fastest way I could think of to do this was to defer all functions found with `resolver.getSymbolsWithAnnotation` and then call `resolver.getSymbolsWithAnnotation` in `finish` and generate everything there.

Not sure if that's frowned upon in KSP, so I'm also working on a solution that generates a file per function so that nothing needs to be cached. The difficulty there is naming the generated file because it's easy to end up with multiple files named the same.